aspnetcore/blazor/javascript-interoperability/import-export-interop.md

```diff
@@ -0,0 +1,370 @@
+
+The module name in the `[JSImport]` attribute and the call to load the module in the component with `JSHost.ImportAsync` syntax should match. The module name must also be unique in the app. When authoring a library for deployment in a NuGet package, we recommend using the NuGet package namespace as a prefix in module names. In the following example, the module name reflects the `Contoso.InteropServices.JavaScript` package and a folder of user message interop classes (`UserMessages`):
+
+```csharp
+[JSImport("getMessage", 
+    "Contoso.InteropServices.JavaScript.UserMessages.CallJavaScript1")]
+```
+
+Export scripts from a standard [JavaScript ES6 module](xref:blazor/js-interop/index#javascript-isolation-in-javascript-modules) either [collocated with a component](xref:blazor/js-interop/index#load-a-script-from-an-external-javascript-file-js-collocated-with-a-component) or placed with other JavaScript static assets in a JS file (for example, `wwwroot/js/{FILENAME}.js`, where JS static assets are maintained in a folder named `js` in the app's `wwwroot` folder and the `{FILENAME}` placeholder is the filename).
+
+In the following example, a JS function named `getMessage` is exported from a collocated JS file that returns a welcome message, "Hello from Blazor!" in Portuguese:
+
+`Pages/CallJavaScript1.razor.js`:
+
+```javascript
+export function getMessage() {
+  return 'Olá do Blazor!';
+}
+```
+
+## Call .NET from JavaScript
+
+This section explains how to call .NET methods from JS.
+
+The following `CallDotNet1` component calls JS that directly interacts with the DOM to render the welcome message string:
+
+* The `CallDotNet` [JS module](xref:blazor/js-interop/index#javascript-isolation-in-javascript-modules) is imported asynchronously from the collocated JS file for this component.
+* The imported `setMessage` JS function is called by `SetWelcomeMessage`.
+* The returned welcome message is displayed by `setMessage` in the UI via the `message` field.
+
+> [!IMPORTANT]
+> In this section's example, JS interop is used to mutate a DOM element *purely for demonstration purposes* after the component is rendered in [`OnAfterRender`](xref:blazor/components/lifecycle#after-component-render-onafterrenderasync). Typically, you should only mutate the DOM with JS when the object doesn't interact with Blazor. The approach shown in this section is similar to cases where a third-party JS library is used in a Razor component, where the component interacts with the JS library via JS interop, the third-party JS library interacts with part of the DOM, and Blazor isn't involved directly with the DOM updates to that part of the DOM. For more information, see <xref:blazor/js-interop/index#interaction-with-the-document-object-model-dom>.
+
+`Pages/CallDotNet1.razor`:
+
+```razor
+@page "/call-dotnet-1"
+@using System.Runtime.InteropServices.JavaScript
+
+<h1>
+    .NET JS <code>[JSImport]</code>/<code>[JSExport]</code> Interop 1 (Call .NET)
```

Comment:
"Interop 2" maybe ?

Reply:
well, I don't know what that number means here. There is also `CallDotNet1`

Reply:
This is numbering the example, not the type of interop. I could add the word "Example" in there.

Reply:
These are named to not conflict 💥 either in a dev's local test app or in the snippet sample app. It's explained to readers here ...

https://learn.microsoft.com/en-us/aspnet/core/blazor/fundamentals/?view=aspnetcore-6.0#razor-components

Reply:
I see the #2 examples now. I guess it's OK. I was just thinking what is the scope of the numbering (as a reader)

Reply:
Could we do header for start of example #2 ?

Reply:
"Example" stuck in there is a good idea. I do that quite a bit around the docs.

Reply:
> Could we do header for start of example 2 ?

It's in a separate section with the heading, "*Use of a single JavaScript module across components*."